### PR TITLE
Add explainer for scope_extensions member

### PR DESCRIPTION
**Pull request by alancutter**

### CODE REVIEW

File: scope_extensions-explainer.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  as one contiguous web app.\
+  E.g. a site may span `example.com`, `example.co.uk` and `support.example.com`.
+
+- Allow web apps to capture user navigations to sites they are affiliated with.\
+  E.g. "News Aggregator App" capturing links navigations to examplenewssite.com.
+
+## Background
+
+Web app scope (defined by the `scope` field) is currently used for:
+1. Constraining urls like start_url, file handlers, share target.
```
Comment (tomayac) [inline scope_extensions-explainer.md]:
```suggestion
1. Constraining URLs appearing in manifest members like `start_url`, `file_handlers`, or `share_target`.
```

File: scope_extensions-explainer.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+other origins given agreement between the web app's primary origin and the
+associated origins.
+
+## Proposal
+
+1. Add a `scope_extensions` member to the web app manifest specifying a list of
+   origin patterns to associate with.
+
+   Example manifest located at `https://example.com/manifest.webmanifest`:
+   ```
```
Comment (tomayac) [inline scope_extensions-explainer.md]:
```suggestion
   ```json
```

File: scope_extensions-explainer.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+     "display": "standalone",
+     "start_url": "/index.html",
+     "scope_extensions": [
+       {"origin": "*.example.com"},
+       {"origin": "example.co.uk"},
+       {"origin": "*.example.co.uk"}
+     ]
+   }
+   ```
+   In this example the Example app is extending its app scope to all its
```
Comment (tomayac) [inline scope_extensions-explainer.md]:
```suggestion
   In this example the "Example" app is extending its app scope to all its
```

File: scope_extensions-explainer.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+   subdomains along with its .co.uk site and its subdomains.
```
Comment (tomayac) [inline scope_extensions-explainer.md]:
```suggestion
   subdomains along with its `.co.uk` site and its subdomains.
```

File: scope_extensions-explainer.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+1. Specify a `web-app-origin-association.json` file format that must be located
```
Comment (tomayac) [inline scope_extensions-explainer.md:55]: This should probably be aligned with https://web.dev/pwa-url-handler/#the-web-app-origin-association-file:

```json
{
  "web_apps": [
    {
      "manifest": "https://music.example.com/manifest.json",
      "details": {
        "paths": ["/*"],
        "exclude_paths": ["/internal/*"]
      }
    }
  ]
}
```

File: scope_extensions-explainer.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+   at `https://<associated origin>/.well-known/web-app-origin-association.json`
+   on the associated origin's domain. This specifies the set of web apps that
+   may include it as a scope extension keyed on each web app's identifier.
+
+   Example association file located at
+   `https://example.co.uk/.well-known/web-app-origin-association.json`:
+   ```
```
Comment (tomayac) [inline scope_extensions-explainer.md]:
```suggestion
   ```json
```

File: scope_extensions-explainer.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    - Has an origin that matches one of the origin patterns in the manifest's
+      `scope_extensions` list.
+    - Has an origin with a valid
+      `<origin>/.well-known/web-app-origin-association.json` association file
+      with an association entry keyed by the web app's identifier.
+    - Matches an `include_paths` entry in the association entry.
+    - Does not match as `exclude_paths` entry in the association entry.
+
+1. Replace the constraint on manifest URLs that are bound by scope to instead be
+   bound by the extended scope with one exception; `start_url` remains bound by
```
Comment (tomayac) [inline scope_extensions-explainer.md]:
```suggestion
   bound by the extended scope—with one exception: `start_url` remains bound by
```

**Review by tomayac — COMMENTED** [review]
Just a passer-by's comments. I have just written the post on PWAs as URL handlers.

File: scope_extensions-explainer.md (hunk 5 above)
Comment (alancutter) [reply scope_extensions-explainer.md:55]: Using the manifest as an identifier isn't compatible with where web app identifiers are headed. I figured if that needed changing then everything is up for grabs to change as well but don't feel strongly about this.

Comment (alancutter) [reply scope_extensions-explainer.md:55]: Context: https://github.com/philloooo/pwa-unique-id/blob/main/explainer.md#recommended-1-start_url_origin--specified-id

Comment (tomayac) [reply scope_extensions-explainer.md:55]: Aware of the context. Just if the file is named the same, it shouldn't be incompatible with the other proposal (well noting that everythig, this and the other proposal, is still in flux). Maybe just add a remark that calls the reader's attention to the other proposal, as they [have done](https://github.com/WICG/pwa-url-handler/blob/main/explainer.md#:~:text=Web%20apps%20can%20be%20identified%20by%20their%20manifest%20URL%20currently%20before%20a%20unique%20identifier%20is%20standardized.).

Comment (alancutter) [reply scope_extensions-explainer.md:55]: I'm hoping we can avoid maintaining backwards compatibility with something that hasn't shipped to the general web.

Adding context SGTM, will update.

Comment (tomayac) [reply scope_extensions-explainer.md:55]: Perfect, thanks! As I said, well noted that everything is in flux :-)

Comment (alancutter) [reply scope_extensions-explainer.md:55]: Done.

**Comment by mgiuca** [comment]
rslgtm

**Pull request merged**